custom_components/pyscript/state.py

```diff
@@ -115,12 +115,12 @@ async def update(cls, new_vars, func_args):
     @classmethod
     def notify_var_get(cls, var_names, new_vars):
         """Return the most recent value of a state variable change."""
-        notify_vars = {}
+        notify_vars = new_vars
         for var_name in var_names if var_names is not None else []:
             if var_name in cls.notify_var_last:
                 notify_vars[var_name] = cls.notify_var_last[var_name]
-            elif var_name in new_vars:
-                notify_vars[var_name] = new_vars[var_name]
+            # elif var_name in new_vars:
+            #     notify_vars[var_name] = new_vars[var_name]
             elif 1 <= var_name.count(".") <= 2 and not cls.exist(var_name):
                 notify_vars[var_name] = None
```

Comment:
This line is going to need some modifications.  I still haven't fully though this through.  The motivation is to avoid exceptions due to undefined variables (eg: undefined state variable or attribute) by setting all variables that are not found in `cls.notify_var_last` or `new_vars` to `None`, so they don't generate an exception when the expression is evaluated.  The logic is bit trickier now: if, say, `domain.entity.attr` is in `var_names`, and `domain.entity` has been set to its `StateVar`, then there's no need to set `domain.entity.attr` to `None` is `attr` is defined in `StateVar`.

Not sure if that makes sense.  I'm happy to deal with this after merge.